scene/animation/animation_player.h

```diff
@@ -394,46 +148,42 @@ class AnimationPlayer : public Node {
 	void set_autoplay(const String &p_name);
 	String get_autoplay() const;
 
-	void set_reset_on_save_enabled(bool p_enabled);
-	bool is_reset_on_save_enabled() const;
-
-	void set_process_callback(AnimationProcessCallback p_mode);
-	AnimationProcessCallback get_process_callback() const;
-
-	void set_method_call_mode(AnimationMethodCallMode p_mode);
-	AnimationMethodCallMode get_method_call_mode() const;
-
-	void set_audio_max_polyphony(int p_audio_max_polyphony);
-	int get_audio_max_polyphony() const;
-
 	void set_movie_quit_on_finish_enabled(bool p_enabled);
 	bool is_movie_quit_on_finish_enabled() const;
 
-	void seek(double p_time, bool p_update = false);
-	void seek_delta(double p_time, double p_delta);
+	void seek(double p_time, bool p_update = false, bool p_update_only = false);
+
 	double get_current_animation_position() const;
 	double get_current_animation_length() const;
 
-	void advance(double p_time);
-
-	void set_root(const NodePath &p_root);
-	NodePath get_root() const;
-
-	void clear_caches(); ///< must be called by hand if an animation was modified after added
-
 	void get_argument_options(const StringName &p_function, int p_idx, List<String> *r_options) const override;
 
-#ifdef TOOLS_ENABLED
-	Ref<AnimatedValuesBackup> backup_animated_values(Node *p_root_override = nullptr);
-	Ref<AnimatedValuesBackup> apply_reset(bool p_user_initiated = false);
-	bool can_apply_reset() const;
-#endif
-
+#ifndef DISABLE_DEPRECATED
+	enum AnimationProcessCallback {
+		ANIMATION_PROCESS_PHYSICS,
+		ANIMATION_PROCESS_IDLE,
+		ANIMATION_PROCESS_MANUAL,
+	};
+	enum AnimationMethodCallMode {
+		ANIMATION_METHOD_CALL_DEFERRED,
+		ANIMATION_METHOD_CALL_IMMEDIATE,
+	};
+	void _set_process_callback_bind_compat_80813(AnimationProcessCallback p_mode);
+	AnimationProcessCallback _get_process_callback_bind_compat_80813() const;
+	void _set_method_call_mode_bind_compat_80813(AnimationMethodCallMode p_mode);
+	AnimationMethodCallMode _get_method_call_mode_bind_compat_80813() const;
+	void _set_root_bind_compat_80813(const NodePath &p_root);
+	NodePath _get_root_bind_compat_80813() const;
+	void _seek_bind_compat_80813(double p_time, bool p_update = false);
+
+	static void _bind_compatibility_methods();
+#endif // DISABLE_DEPRECATED
 	AnimationPlayer();
 	~AnimationPlayer();
 };
-
+#ifndef DISABLE_DEPRECATED
 VARIANT_ENUM_CAST(AnimationPlayer::AnimationProcessCallback);
 VARIANT_ENUM_CAST(AnimationPlayer::AnimationMethodCallMode);
+#endif // DISABLE_DEPRECATED
```

Comment:
And would be nice to preserve the line break before this block (same for the animation tree).